src/odm_sharing/private/queries.py

```diff
@@ -1,14 +1,326 @@
-from dataclasses import dataclass
-from typing import Dict
+import re
+from collections import defaultdict
+from dataclasses import dataclass, field
+from functools import partial
+from typing import Dict, List, Tuple
+# from pprint import pprint
 
+from functional import seq
+
+from odm_sharing.private.common import OrgName, TableName
 from odm_sharing.private.rules import RuleId
+from odm_sharing.private.stdext import StrEnum, sorted_dict
+from odm_sharing.private.utils import dqt
+
+from odm_sharing.private.trees import (
+    ALL_LIT,
+    Node,
+    NodeKind,
+    Op,
+    RangeKind,
+    RuleTree,
+    parse_op,
+)
+
+
+Args = List[str]
```

Comment:
I could do "arguments" but it's so long. I think "args" is a very common short form. I think "parameters" is more confusing since these are the values passed/arguments passed in to the parameters.